### ISSUE TITLE
`calicoctl node checksystem` fails minimum systems requirements on arm64

### ISSUE THREAD

**Issue by JeffreyUrban** [projectcalico/calicoctl#2204]
## Desired Behavior
_(desired, not expected, since state of calico arm64 support is unclear)_
---
`calicoctl node checksystem` should pass on a Linux distribution on arm64 
OR 
There is a method to obtain and add modules (`ipt_set`, `xt_icmp`, `xt_icmp6`, `xt_rpfilter`) so it will pass.
OR 
There is guidance on whether is possible to run calico on arm64 without modules (`ipt_set`, `xt_icmp`, `xt_icmp6`, `xt_rpfilter`), and any constraints or cautions.

## Current Behavior
For each <module> of (`ipt_set`, `xt_icmp`, `xt_icmp6`, `xt_rpfilter`), 
`calicoctl node checksystem` reports:
```
    WARNING: Unable to detect the <module> module as Loaded/Builtin module or lsmod
    <module>    FAIL

```
## Possible Solutions
Determine the calico options or features that require these kernel modules and determine if is is practicable to use calico on arm64 without using these features or options.
OR
Determine how to obtain and install (`ipt_set`, `xt_icmp`, `xt_icmp6`, `xt_rpfilter`) on an arm64 Linux distribution.
OR 
Document why these are necessary for calico, as a caution for arm64 users.
OR
Hear about others using calico successfully on arm64 in production.

## Steps to Reproduce (for bugs)
```
wget https://github.com/projectcalico/calicoctl/releases/download/v3.16.3/calicoctl-linux-arm64
chmod a+x calicoctl-linux-arm64
./calicoctl node checksystem

```
## Context
I'm trying to determine if calico can be a stable option for arm64 dev boards, and if it can be feature complete, or limited.

Kernel modules (`ipt_set`, `xt_icmp`, `xt_icmp6`, `xt_rpfilter`) are listed as requirements for calico in:
https://docs.projectcalico.org/getting-started/kubernetes/requirements
https://docs.projectcalico.org/getting-started/openstack/requirements
https://docs.projectcalico.org/getting-started/bare-metal/requirements
...with: _"Tip: If you are using one of the recommended distributions, you will already satisfy these."_

x86-64 processor is also a requirement, but there are arm64 builds, such as:
https://github.com/projectcalico/calicoctl/releases/download/v3.16.3/calicoctl-linux-arm64

Searching the Internet, I can't find any mention of the source or purpose of (`ipt_set`, `xt_icmp`, `xt_icmp6`, `xt_rpfilter`), so the implications of these failures are not clear.

Commentary on this issue suggests that these may be required by calico in error:
https://github.com/kubernetes-sigs/kubespray/issues/6289

## Your Environment

* calicoctl version (e.g. `calicoctl version`):
https://github.com/projectcalico/calicoctl/releases/download/v3.16.3/calicoctl-linux-arm64
* calico/node version (e.g. `docker run --rm calico/node versions`)
N/A
* Orchestrator type and version (e.g. kubernetes, mesos, rkt):
N/A
* Operating System and version:
rk3328-sd-friendlycore-bionic-5.4-arm64-20200707 (Ubuntu Core)
rk3328-sd-friendlywrt-5.4-20200904 (OpenWRT)
Armbian_20.08.1_Nanopi-r2s_buster_current_5.8.6 (Debian)
ubuntu-20.04.1-preinstalled-server-arm64+raspi (Ubuntu Server)

* Link to your project (optional):
N/A

**Comment by JeffreyUrban** [comment]
Someone on the forums reported these same failures with amd64 (on Debian 10), so may not be specific to arm64.
kernel:
Linux X 4.19.0-11-amd64 #1 SMP Debian 4.19.146-1 (2020-09-17) x86_64 GNU/Linux
https://calicousers.slack.com/archives/C0BCA117T/p1602063050064900

**Comment by JeffreyUrban** [comment]
Appears here is where checks for these modules were introduced:
Issue: https://github.com/projectcalico/calicoctl/issues/1444
Pulls: https://github.com/projectcalico/calico/pull/3076/files, https://github.com/projectcalico/calicoctl/pull/2090/files

**Comment by caseydavenport** [comment]
Yeah, I think the checksystem command is itself pretty outdated, and IIRC it sometimes checks more dependencies than you actually need.

We need to audit the dependencies in that command and bring it up to speed with modern Calico. For now though, I would just ignore its output. Most distros have what you need anyway.
